src/mutations/collection/deleteCollections.ts

```diff
@@ -0,0 +1,39 @@
+import { ForbiddenError, UserInputError } from 'common/errors'
+import { fromGlobalId } from 'common/utils'
+import { MutationToDeleteCollectionsResolver } from 'definitions'
+
+const resolver: MutationToDeleteCollectionsResolver = async (
+  _,
+  { input: { ids } },
+  { dataSources: { collectionService }, viewer }
+) => {
+  if (!viewer.id) {
+    throw new ForbiddenError('Viewer has no permission')
+  }
+  if (ids.length === 0) {
+    return false
+  }
+
+  const unpacked = ids.map((id) => fromGlobalId(id))
+  const types = unpacked.map((d) => d.type)
+
+  if (types.some((type) => type !== 'Collection')) {
+    throw new UserInputError('Invalid collection ids')
+  }
+
+  const collectionIds = unpacked.map((d) => d.id)
+
+  try {
+    return await collectionService.deleteCollections(collectionIds, viewer.id)
+  } catch (e: any) {
+    if (e.message === 'Collection not found') {
```

Comment:
shall we throw the error directly in `deleteCollections`

Reply:
The point I thow simple Error first instead of errors in `src/common/errors.ts` is that those errors are subclass of  `GraphQLError`  and are used for api layer, not service layer.

Reply:
I changed my mind to use Errors in `common/errors.ts`, which is more practical  and  keep consistent  with other place